ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/jdbc/JDBCExpandExpressionsRule.java

```diff
@@ -178,6 +177,9 @@ protected static class RexTransformIntoOrAndClause extends RexShuttle {
     @Override
     public RexNode visitCall(RexCall inputCall) {
       RexNode node = super.visitCall(inputCall);
+      if (!RexUtil.isFlat(node)) {
+        node = RexUtil.flatten(rexBuilder, node);
+      }
```

Comment:
If we flatten the call here aren't we risking to create a non-flat expression during the IN to OR transformation below?

Reply:
In fact, I was thinking that it may be better to flatten after the transformation but if you are saying that it's not necessary that's fine with me.
```java
public RexNode visitCall(RexCall inputCall) {
      RexNode node = super.visitCall(inputCall);
      if (node instanceof RexCall) {
        RexCall call = (RexCall) node;
        switch (call.getKind()) {
          case IN:
            node = transformIntoOrAndClause(rexBuilder, call);
          default:
            break;
        }
      }
      return RexUtil.isFlat(node) ? node : RexUtil.flatten(rexBuilder, node);
    }
```

Reply:
Although it is doing the same thing, this indeed looks safer, so I have included it in the latest commit.